src/cache.rs

```diff
@@ -547,6 +547,12 @@ impl Cache {
         &mut self.files
     }
 
+    /// Get a mutable reference to the cached term roots
+    /// (used by the language server to invalidate previously parsed entries)
+    pub fn terms_mut(&mut self) -> &mut HashMap<FileId, (RichTerm, EntryState)> {
+        &mut self.terms
+    }
+
```

Comment:
I don't really like the idea of exposing publicly the underlying hashmap. That's ok for now if it's a blocker for you, because it seems the cache is not handling file changes correctly, but let us maybe open an issue to fix the actual problem and remember to remove this once this is not needed anymore.

Reply:
Yes I see the problems with this, `pub` is remarkably open.
Alternatively we could have an `invalidate` method, but if we start that and want to use the cache term structures in future analysis we start adding a bunch of methods just for the lsp
If we don't want to do either I have to reimplement most parts of the cache (which might be the most applicable thing anyway if it turns out that the processing would require too much work on the existing nickel module)

Yes, let me open an issue..